crypto/rand/drbg_rand.c

```diff
@@ -341,9 +341,9 @@ int ctr_init(RAND_DRBG *drbg)
         AES_set_encrypt_key(df_key, drbg->strength, &ctr->df_ks);
 
         drbg->min_entropylen = ctr->keylen;
-        drbg->max_entropylen = DRBG_MAX_LENGTH;
+        drbg->max_entropylen = DRBG_MINMAX_FACTOR * drbg->min_entropylen;
```

Comment:
I would like to see some check to verify that it's not larger than what the NIST documentation has, which can probably be a compile time error.

Reply:
Really? Normally, I am a great friend of static asserts, but in this case I would say that it is overkill: Both factors, `DRBG_MINMAX_FACTOR` and `drbg->min_entropylen` are hardcoded and not user input. And we are talking about values like `128 * 256` in the worst case. So I would say, since the 16bit era is history, we can trust the compiler to calculate this value correctly.  ;-)